### ISSUE TITLE
Issue 006: fine dataset loader を generic center NPZ loader に置き換える

### ISSUE THREAD

**Issue by ozw4** [ozw4/seisai#339]
# Issue 006: fine dataset loader を generic center NPZ loader に置き換える

## 背景

Issue 005 で `center_npz_files` が fine config に入る。しかし dataset 実装はまだ physics 固定である。

現状:

```python
from seisai_engine.pipelines.fbpick.common import load_physics_npz

_load_physics_window_centers_for_info(...)
FineInferenceGatherWindowsDataset(physics_npz_files=...)
build_train_dataset(..., physics_npz_files=...)
build_raw_infer_dataset(..., physics_npz_files=...)
```

さらに `_load_physics_window_centers_for_info()` は `physical_center_i != -1` を hard gate しており、coarse center を直接使う場合に不要な physics 依存になる。

## 目的

fine dataset の window center loading を generic NPZ + key mapping にする。

```text
external NPZ:
  physics npz / coarse npz / future center npz

config:
  window_center.npz_key
  window_center.fallback_npz_key
  window_center.valid_mask_npz_key

internal:
  centers: int64 vector
  center_valid_mask: bool vector
```

## 非目的

- fine inference artifact の schema 変更は Issue 007。
- `physics_npz_files` alias 削除はしない。
- training の `require_fb_inside` policy 変更は別 issue。

## 変更対象

```text
packages/seisai-engine/src/seisai_engine/pipelines/fbpick/fine/build_dataset.py
packages/seisai-engine/src/seisai_engine/pipelines/fbpick/fine/train.py
packages/seisai-engine/src/seisai_engine/pipelines/fbpick/fine/infer.py
packages/seisai-engine/src/seisai_engine/pipelines/fbpick/fine/__init__.py
packages/seisai-engine/tests/test_fbpick_fine_dataset.py  # なければ既存 fine dataset tests に追加
```

Issue 001 の `load_npz_payload()` / alignment helper を使う。

### 1. loader rename / generic 化

以下を置き換える。

```python
_load_physics_window_centers_for_info(...)
```

新規:

```python
def _load_window_centers_for_info(
    center_npz_path: str,
    info,
    *,
    npz_key: str = 'physics_window_center_i',
    fallback_npz_key: str | None = None,
    valid_mask_npz_key: str | None = None,
    invalid_center_value: int = -1,
    guard_npz_key: str | None = None,
    guard_invalid_value: int = -1,
) -> tuple[np.ndarray, np.ndarray | None]:
    ...
```

既存 private function 名は test 互換が必要なら wrapper として残す。

```python
def _load_physics_window_centers_for_info(...):
    return _load_window_centers_for_info(..., guard_npz_key='physical_center_i')
```

ただし main path では generic 関数を使う。

### 2. generic NPZ loading

`load_physics_npz()` は使わない。

```python
center_payload = load_npz_payload(center_npz_path, payload_name='fine center npz')
alignment = read_npz_alignment(center_payload, payload_name='fine center npz')
validate_npz_alignment_against_info(alignment, info, payload_name='fine center npz')
```

これにより `.coarse.npz` のように physics required keys を持たない payload も読める。

### 3. center vector selection

`window_center.npz_key` / `fallback_npz_key` を使って center vector を選ぶ。

```python
selected_key, selected = select_npz_array(...)
centers = coerce_integer_index_vector(
    selected,
    key=selected_key,
    length=n_traces,
    n_samples_orig=n_samples_orig,
    invalid_value=invalid_center_value,
    allow_invalid=True,
)
```

`centers` は `int64` で返し、呼び出し側は既存通り `int64` / pad 時に処理する。

### 4. valid mask selection

`valid_mask_npz_key` がある場合:

```python
valid_mask = coerce_bool_mask_vector(payload[valid_mask_npz_key], ...)
```

`valid_mask_npz_key` が null の場合:

```python
valid_mask = centers != invalid_center_value
```

### 5. optional guard

physics legacy と同じ hard gate が必要な場合だけ使える optional guard を追加する。

```yaml
window_center:
  npz_key: physics_window_center_i
  valid_mask_npz_key: physics_window_valid_mask
  guard_npz_key: physical_center_i
  guard_invalid_value: -1
```

ただし Issue 005 時点の `FineWindowCenterCfg` には `guard_npz_key` がない。実装を最小にするなら、この issue で `FineWindowCenterCfg` に以下を追加する。

```python
guard_npz_key: str | None = None
guard_invalid_value: int = -1
invalid_center_value: int = -1
```

default は後方互換を考えて慎重に決める。

推奨 default:

```yaml
window_center:
  invalid_center_value: -1
  guard_npz_key: null
  guard_invalid_value: -1
```

理由: `physics_window_valid_mask` がすでに window validity を表すため、`physical_center_i` hard gate は generic center source では不要。既存挙動との差分が心配なら examples だけに `guard_npz_key: physical_center_i` を明示してもよい。

### 6. class / function argument rename

可能な範囲で public-ish names を generic にする。

変更前:

```python
FineInferenceGatherWindowsDataset(physics_npz_files=...)
build_train_dataset(..., physics_npz_files=...)
build_labeled_infer_dataset(..., physics_npz_files=...)
build_raw_infer_dataset(..., physics_npz_files=...)
```

変更後:

```python
FineInferenceGatherWindowsDataset(center_npz_files=...)
build_train_dataset(..., center_npz_files=...)
build_labeled_infer_dataset(..., center_npz_files=...)
build_raw_infer_dataset(..., center_npz_files=...)
```

後方互換のため、function signature に `physics_npz_files` alias を残してもよい。推奨は以下。

```python
def build_raw_infer_dataset(*, segy_files, center_npz_files=None, physics_npz_files=None, ...):
    center_npz_files = _resolve_center_npz_files(center_npz_files, physics_npz_files)
```

error message は new field を優先する。

### 7. train.py / infer.py 呼び出し変更

Issue 005 で `typed.paths.center_npz_files` が入っているので、dataset build では canonical field を渡す。

```python
center_npz_files=list(typed.paths.center_npz_files)
```

`typed.paths.physics_npz_files` ではなく canonical を使う。

## テスト要件

候補テスト:

```python
def test_load_window_centers_for_info_reads_physics_npz_with_configured_key() -> None: ...

def test_load_window_centers_for_info_reads_coarse_npz_without_physics_required_keys() -> None: ...

def test_load_window_centers_for_info_uses_fallback_key() -> None: ...

def test_load_window_centers_for_info_uses_valid_mask_key() -> None: ...

def test_load_window_centers_for_info_no_longer_requires_physical_center_i_by_default() -> None: ...

def test_fine_raw_infer_dataset_accepts_center_npz_files_alias() -> None: ...
```

coarse-like test payload は以下の最小 keys でよい。

```text
dt_sec
n_samples_orig
n_traces
trace_indices
coarse_pick_i
coarse_center_valid_mask
```

ただし `read_npz_alignment()` が `ffid_values` 等を要求しない設計なら、center loader は上記だけでよい。もし alignment helper が full fbpick metadata を読む設計なら、それに合わせる。

## 受け入れ条件

- fine dataset が `center_npz_files` を使う。
- `.physics.npz` と `.coarse.npz` のどちらでも center vector を読める。
- `load_physics_npz()` は generic center loading path では使われない。
- `physical_center_i` は default では required ではない。
- 既存 `physics_npz_files` config も動く。

## 推奨確認コマンド

```bash
pytest packages/seisai-engine/tests -q -k 'fine and dataset'
pytest packages/seisai-engine/tests/test_fbpick_physics_window_constraint.py -q
```
